sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_distillation/distillation_job.py

```diff
@@ -0,0 +1,87 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from marshmallow import fields
+
+from azure.ai.ml._schema._distillation.distillation_types import PromptSettingsSchema, TeacherModelSettingsSchema
+from azure.ai.ml._schema._finetuning.finetuning_vertical import FineTuningVerticalSchema
+from azure.ai.ml._schema.core.fields import (
+    ArmVersionedStr,
+    LocalPathField,
+    NestedField,
+    RegistryStr,
+    StringTransformedEnum,
+    UnionField,
+)
+from azure.ai.ml._schema.job import BaseJobSchema
+from azure.ai.ml._schema.job.input_output_entry import DataInputSchema, ModelInputSchema
+from azure.ai.ml._schema.job.input_output_fields_provider import OutputsField
+from azure.ai.ml._schema.job_resource_configuration import JobResourceConfigurationSchema
+from azure.ai.ml._schema.workspace.connections import ServerlessConnectionSchema, WorkspaceConnectionSchema
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.constants import DataGenerationTaskType, DataGenerationType, JobType
+from azure.ai.ml.constants._common import AzureMLResourceType
+
+
+@experimental
+class DistillationJobSchema(FineTuningVerticalSchema, BaseJobSchema):
+    type = StringTransformedEnum(required=True, allowed_values=JobType.DISTILLATION)
+    data_generation_type = StringTransformedEnum(
+        allowed_values=[DataGenerationType.LabelGeneration, DataGenerationType.DataGeneration],
+        required=True,
+    )
+    data_generation_task_type = StringTransformedEnum(
+        allowed_values=[
+            DataGenerationTaskType.NLI,
+            DataGenerationTaskType.NLU_QA,
+            DataGenerationTaskType.CONVERSATION,
+            DataGenerationTaskType.MATH,
+            DataGenerationTaskType.SUMMARIZATION,
+        ],
+        casing_transform=str.upper,
+        required=True,
+    )
+    teacher_model_endpoint_connection = UnionField(
+        [NestedField(WorkspaceConnectionSchema), NestedField(ServerlessConnectionSchema)], required=True
```

Comment:
Could we make it optional now? and make it required when we have service support

Reply:
We would still require the "name" portion of the endpoint name since we need to know what endpoint to use so it can't be optional. It could be changed to where we require just the "teacher_model_endpoint_name" but me and Miseon talked about it decided it would be best to keep it as ServerlessConnection to minimize users having to change their sdk code and yaml files once we use the ServerlessConnection in its entirety